### ISSUE TITLE
Metal: Failed Assertion  / Crash when trying to create arrays in double-precision

### ISSUE THREAD

**Issue by johnjmolina** [jax-ml/jax#16435]
### Description

There seems to be an issue with casting for the edge case of size zero arrays.

```python
from jax import config
config.update("jax_enable_x64", True)
import jax.numpy as jnp
import numpy as np

jnp.array([1.0])           # OK : Array([1.], dtype=float64)
jnp.array(np.double(1.0))  # OK : Array(1., dtype=float64)
jnp.array(jnp.double(1.0)) # Crash
jnp.array(1.0)             # Crash  
```
The last two commands will crash the kernel
```
Metal device set to: Apple M1

systemMemory: 16.00 GB
maxCacheSize: 5.33 GB

/AppleInternal/Library/BuildRoots/c2cb9645-dafc-11ed-aa26-6ec1e3b3f7b3/Library/Caches/com.apple.xbs/Sources/MetalPerformanceShadersGraph/mpsgraph/MetalPerformanceShadersGraph/Core/Files/MPSGraphExecutable.mm:1377: failed assertion `Incompatible element type for parameter at index 0, mlir module expected element type f64 but received f32'
```

### What jax/jaxlib version are you using?

jax 0.4.11, jaxlib 0.4.10, jax-metal 0.0.2

### Which accelerator(s) are you using?

Apple GPU

**Comment by jakevdp** [comment]
Just to confirm, this is on the Apple metal backend, yes?

**Comment by hawkinsp** [comment]
I think the Metal plugin simply doesn't support f64, but it should report it more gracefully.

**Comment by kulinseth** [comment]
@hawkinsp , that's right. Metal doesn't have FP64 support, we will add support to fail more gracefully.

**Comment by kulinseth** [comment]
@johnjmolina , this issue is not related to Empty shapes. We don't have support for FP64 and we will add error-checks early to exit gracefully.

**Comment by johnjmolina** [comment]
Thanks for the prompt reply, I understand the issue now.

**Comment by johannahaffner** [comment]
> @hawkinsp , that's right. Metal doesn't have FP64 support, we will add support to fail more gracefully.

Are you planning on adding FP64 support? I would be really keen on that.

**Comment by johannahaffner** [comment]
@shuhand0 thank you for letting me know.

**Comment by xuhengdada** [comment]
Is it possible for jax to support FP64 using libraries like: https://github.com/philipturner/metal-float64 ?

**Comment by jakevdp** [comment]
The last release of jax-metal was over a year ago now, and so it seems pretty clear that the team behind it is no longer working on it. We've removed mentions of jax-metal from JAX's installation docs: at this point I think it's probably best for users to consider the jax-metal project dead, and to focus on using supported backends.
